### PR TITLE
Fix names of some reference functions

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88117
* #88116
* __->__ #88115

The `__name__` field of some binary reference functions was wrong. We
fix this to be consistent with unary reference functions. In the future,
we should probably make the binary reference wrapper return a wrapper
itself to avoid all those calls to `partial`.

This change helps performing some homogeneous treatment of functions by
their name.

cc @ezyang @mruberry @ngimel @Lezcano @peterbell10 @fdrocha

### CODE REVIEW

**Comment by ezyang** [comment]
See also https://github.com/pytorch/pytorch/pull/79178

**Comment by ezyang** [comment]
Well, now that you bashed out all the partials, I guess we can take it, but I probably would have added the decorator variant first haha

File: torch/_refs/__init__.py [hunk 1]
```diff
@@ -944,47 +949,70 @@ def add(
 
 
 # TODO: add docstring
-atan2 = _make_elementwise_binary_reference(
-    prims.atan2,  # type: ignore[has-type]
+@partial(
```
Comment (mruberry) [inline torch/_refs/__init__.py:952]: I don't love the pattern of `@partial` for this and then having to basically redefine the prim in a wrapper. Are there some other options?

**Comment by lezcano** [comment]
@ezyang vim macros FTW :D

File: torch/_refs/__init__.py (hunk 1 above)
Comment (lezcano) [reply torch/_refs/__init__.py:952]: I'll remove all the partials in this PR as well.

File: torch/_prims_common/__init__.py [hunk 2]
```diff
@@ -1576,6 +1576,27 @@ def mask_tensor(mask: TensorLikeType, t: TensorLikeType):
         return torch.where(mask, t, 0)
 
 
+def get_aten_op(fn: Callable, name: str):
+    """
+    Given the __module__ of reference and its name, it returns
```
Comment (mruberry) [inline torch/_prims_common/__init__.py:1581]: Isn't there a reference->ATen mapping that this should be using?

File: torch/_refs/__init__.py [hunk 3]
```diff
@@ -860,54 +855,59 @@ def trunc(a):
...
-    )
-    def _ref(
-        a: Union[Tensor, NumberType],
-        b: Union[Tensor, NumberType],
-    ) -> Tensor:
-        if not supports_lhs_python_scalar and isinstance(a, Number):
-            raise ValueError(
-                "Received a lhs Python scalar to an elementwise binary operation that does not accept lhs scalars!"
-            )
+    def inner(prim: Callable):
```
Comment (mruberry) [inline torch/_refs/__init__.py:866]: What's going on with this change?

File: torch/_prims_common/__init__.py (hunk 2 above)
Comment (lezcano) [reply torch/_prims_common/__init__.py:1581]: I have no idea. I don't think such a thing exists. Perhaps @ezyang knows?

File: torch/_refs/__init__.py (hunk 3 above)
Comment (lezcano) [reply torch/_refs/__init__.py:866]: This just aligns `_make_elementwise_binary_reference` with `_make_elementwise_unary_reference` in that, given some parameters, it returns a decorator.

Comment (lezcano) [reply torch/_refs/__init__.py:866]: I also moved from if nots to checks, improve the error messages (priting the correct name) and I set the name of the `ref`

File: torch/_prims_common/__init__.py (hunk 2 above)
Comment (ezyang) [reply torch/_prims_common/__init__.py:1581]: Whenever a ref is registered as a decomposition for an aten op (`register_decomposition`) that also sets up a ref to aten mapping. We don't publish this API but the info is there inside a hash table.

Comment (lezcano) [reply torch/_prims_common/__init__.py:1581]: Yep, that's what I thought. This function is meant to be called in order to then register the relevant decomposition later on, so we can't make use of that.

**Review by mruberry — APPROVED** [review]
OK

**Pull request closed without merging**